### DISCUSSION TITLE
Logging the noconsole exception dialog

### DISCUSSION THREAD

**Discussion by caffeinepills** [pyinstaller/pyinstaller#6307] [Help] [answered]
In window mode (no console) if an exception occurs, it pops up with a dialog box showing the exception and interrupts the program. While the traceback is very handy in terms of displaying the error, it is not ideal as the error does not get logged. Is there anyway to disable this popup, or at the very least allow loggers to get the exception that occurs?  

In a previous version (4.2) while it did popup a basic dialog message with no information, the traceback was at least allowed logged to a file. When someone submits bug reports, I like to have them send that log file so I can see a past history of all the exceptions that occurred, as it's not reasonable to expect every user to screenshot every exception dialog that occurs. Just wondering how I can go about accomplishing this?  I tried the `--disable-windowed-traceback` flag, but the dialog still pops up.

EDIT: I am currently trying the 5.0 dev branch.

The popup in question:
![image](https://user-images.githubusercontent.com/17804711/137843191-337a6726-db1b-4e1e-b7a9-81199bf9adb8.png)

**Comment by bwoodsend** [comment]
If you don't want the exception dialog then you need to handle your own errors instead of letting them propagate.
```python
try:
    # your code here
    a = []
    a[1]
except:
    import traceback
    with open("error.log", "w") as f:
        f.write(traceback.format_exc())
    raise SystemExit
```

**Comment by caffeinepills** [comment, marked as answer]
I guess a try block over the whole code could technically work, but not quite what I had in mind. 

I did find a workaround using `sys.excepthook`:
```py
if getattr(sys, 'frozen', False):
    def frozenhook(*exc_info):
        log.msg("".join(traceback.format_exception(*exc_info)))

    sys.excepthook = frozenhook
```

Tested it with a frozen application and it now logs the exceptions.

**Reply by bwoodsend** [reply]
That's quite neat actually because it works for background threads too. Nice find!
